magma_access_gateway_installer/agw_network_configurator.py

```diff
@@ -0,0 +1,237 @@
+        updated_grub_file.close()
+
+    def _update_interfaces_names_in_cloud_init(self):
+        """Changes names of network interfaces in /etc/netplan/50-cloud-init.yaml to ethX."""
+        cloud_init_config_file = "/etc/netplan/50-cloud-init.yaml"
+        with open(cloud_init_config_file, "r") as cloud_init_file:
+            cloud_init_content = yaml.safe_load(cloud_init_file)
+
+        cloud_init_interfaces = list(cloud_init_content["network"]["ethernets"].keys())
+        for network_interface in cloud_init_interfaces:
+            network_interface_index = cloud_init_interfaces.index(network_interface)
+            logger.info(
+                f"Changing interface name in {cloud_init_config_file}. "
+                f"Old interface name: {network_interface} "
+                f"New interface name: eth{network_interface_index}."
+            )
+            cloud_init_content["network"]["ethernets"][
+                f"eth{network_interface_index}"
+            ] = cloud_init_content["network"]["ethernets"][network_interface]
+            del cloud_init_content["network"]["ethernets"][network_interface]
+            cloud_init_content["network"]["ethernets"][f"eth{network_interface_index}"][
+                "set-name"
+            ] = f"eth{network_interface_index}"
+
+        with open(cloud_init_config_file, "w") as cloud_init_file:
+            yaml.dump(cloud_init_content, cloud_init_file)
+
+        self.REBOOT_NEEDED = True
+
+    def _update_grub_cfg(self):
+        """Updates /boot/grub/grub.cfg."""
+        logger.info("Updating /boot/grub/grub.cfg...")
+        check_call(["grub-mkconfig", "-o", "/boot/grub/grub.cfg"])
+        self.REBOOT_NEEDED = True
+
+    @property
+    def _dns_configured(self) -> bool:
+        """Checks whether DNS has already been configured."""
+        with open("/etc/systemd/resolved.conf", "r") as dns_config:
+            for line in dns_config.readlines():
+                if f"DNS={self.DNS_ADDRESSES}" in line:
+                    return True
+        return False
+
+    def _configure_dns(self):
+        """Configures Ubuntu's DNS."""
+        logger.info("Configuring DNS...")
+        try:
+            os.symlink("/var/run/systemd/resolve/resolv.conf", "/etc/resolv.conf")
+        except FileExistsError:
+            os.remove("/etc/resolv.conf")
+            os.symlink("/var/run/systemd/resolve/resolv.conf", "/etc/resolv.conf")
+        original_resolved_conf_file = open("/etc/systemd/resolved.conf", "r")
+        original_resolved_conf_file_content = original_resolved_conf_file.readlines()
+        updated_resolved_conf_file_content = [
+            line.replace(line, f"DNS={self.DNS_ADDRESSES}\n")
+            if line.startswith("#DNS=") or line.startswith("DNS=")
+            else line
+            for line in original_resolved_conf_file_content
+        ]
+        original_resolved_conf_file.close()
+
+        updated_resolved_conf_file = open("/etc/systemd/resolved.conf", "w")
```

Comment:
We could use with like in the` _update_interfaces_names_in_cloud_init` method

Reply:
Done.